### PR TITLE
Updated eslint-plugin-minecraft-linting

### PR DESCRIPTION
**Pull request by SBLMikeDemone**
eslint-plugin-minecraft-linting was updated to catch a few more potential issues here: https://github.com/Mojang/minecraft-scripting-libraries/pull/21 and this PR updates all the samples and ensures they all pass `npm run lint`.

Tested by running the ts-starter-complete-cotta and ensuring my score was reset to 0 when restarting.

### CODE REVIEW

File: howto-gallery/package.json [hunk 1]
```diff
@@ -40,4 +40,4 @@
       "@minecraft/server": "$@minecraft/server"
     }
   }
-}
\ No newline at end of file
+}
```
Comment (SBLMikeDemone) [inline howto-gallery/package.json:43]: I noticed when pulling down this repo and then running `npm run lint` that most (all?) files failed linting due to file endings. I ran `npm run lint -- --fix` and it fixed them all locally but git didn't recognize the difference. Is there something weird going on with line endings?

Comment (frgarc) [reply howto-gallery/package.json:43]: +1, not sure if we are missing some config.

File: ts-starter-complete-cotta/scripts/main.ts [hunk 2]
```diff
@@ -41,7 +49,7 @@ function initializeBreakTheTerracotta() {
   let players = world.getAllPlayers();
 
   for (let player of players) {
-    player.runCommand("scoreboard players set @s score 0");
+    scoreObjective.setScore(player.scoreboardIdentity as ScoreboardIdentity, 0);
```
Comment (frgarc) [inline ts-starter-complete-cotta/scripts/main.ts:52]: ❤️

**Review by frgarc — APPROVED** [review]
LGTM, tx

File: howto-gallery/package.json (hunk 1 above)
Comment (rlandav) [reply howto-gallery/package.json:43]: This repository doesn't have build pipelines, and this package is used more for snippets, so it's possibly that simply this is just stale.

File: ts-starter-complete-cotta/scripts/main.ts (hunk 2 above)
Comment (frgarc) [reply ts-starter-complete-cotta/scripts/main.ts:52]: Ok, minor comment, would be better to do an undefine check here to avoid using `as`

Comment (SBLMikeDemone) [reply ts-starter-complete-cotta/scripts/main.ts:52]: Hm yeah, I was actually confused why/when scoreboardIdentity could be null but I guess if one of the players is a spectator. I'll throw in an undefined check

File: howto-gallery/package.json (hunk 1 above)
Comment (SBLMikeDemone) [reply howto-gallery/package.json:43]: I created this user story to fix the linting issues: [minecraft-scripting-samples fails linting on a fresh clone](https://dev.azure.com/dev-mc/Minecraft/_workitems/edit/1207082)

**Review by frgarc — APPROVED** [review]

**Comment by SBLMikeDemone** [comment]
Updated to 1.2.1 to get rid of a bunch of dependencies (nearly 1000 lines from each package-lock.json woo!)

**Comment by frgarc** [comment]
> Updated to 1.2.1 to get rid of a bunch of dependencies (nearly 1000 lines from each package-lock.json woo!)

was only the version the issue? 😮

**Comment by SBLMikeDemone** [comment]
> was only the version the issue? 😮

@frgarc Ah, no, I should've been more clear. Raph moved a bunch of core build tool dependencies to dev dependencies: https://github.com/Mojang/minecraft-scripting-libraries/pull/22

File: ts-starter-complete-cotta/scripts/main.ts [hunk 3]
```diff
@@ -41,7 +49,10 @@ function initializeBreakTheTerracotta() {
   let players = world.getAllPlayers();
 
   for (let player of players) {
-    player.runCommand("scoreboard players set @s score 0");
+    if (player.scoreboardIdentity === undefined) {
```
Comment (rlandav) [inline ts-starter-complete-cotta/scripts/main.ts:52]: nit: you can simply do `if (!player.scoreboardIdentity) { `

**Review by rlandav — APPROVED** [review]

File: ts-starter-complete-cotta/scripts/main.ts (hunk 3 above)
Comment (frgarc) [reply ts-starter-complete-cotta/scripts/main.ts:52]: This was already simplified. Just merge with latest main.

**Comment by SBLMikeDemone** [comment]
@microsoft-github-policy-service agree company="Skybox Labs"

**Comment by SBLMikeDemone** [comment]
@frgarc After grabbing your changes this PR is significantly smaller! Re-requesting a review post merge :)

File: ts-starter-complete-cotta/scripts/main.ts [hunk 4]
```diff
@@ -1,4 +1,12 @@
-import { world, system, BlockPermutation, EntityInventoryComponent, ItemStack, DisplaySlotId } from "@minecraft/server";
+import {
+  world,
+  system,
+  BlockPermutation,
+  EntityInventoryComponent,
+  ItemStack,
+  DisplaySlotId,
+  ScoreboardIdentity,
```
Comment (frgarc) [inline ts-starter-complete-cotta/scripts/main.ts:8]: I think `ScoreboardIdentity` is not more needed

**Review by frgarc — COMMENTED** [review]
Only one question added

File: ts-starter-complete-cotta/scripts/main.ts (hunk 4 above)
Comment (SBLMikeDemone) [reply ts-starter-complete-cotta/scripts/main.ts]: You're absolutely right! I thought that was just a prettier fix thing. Good catch. Now it's a super tiny PR!

**Review by frgarc — APPROVED** [review]

**Pull request merged**